### DISCUSSION TITLE
latency statistic data is empty

### DISCUSSION THREAD

**Discussion by wannastar** [eProsima/Fast-DDS-monitor#195] [Q&A]
![Uploading image.png…]()

![image](https://github.com/eProsima/Fast-DDS-monitor/assets/42711319/9cdd87ed-3784-4a94-b8b0-1462a5b34aab)

step1:
   compile fastdds with -DFASTDDS_STATISTICS=ON（default is on）and -DPERFORMANCE_TESTS=ON
step2:
   ./LatencyTest publisher   -f payloads_demands.csv --domain 0 -s 1000000
   ./LatencyTest subscriber -f payloads_demands.csv --domain 0
step3:
   configure fastddsmonitor to introspect FASTDDS LATENCY like picture1, but the graph is always empty.

**Comment by rsanchez15** [comment]
Hi @wannastar 

As you mentioned, FASTDDS_STATISTICS is set to ON by default in latest versions of Fast DDS. However, this does not imply that the internal statistics are enabled, as you should explicitly enable each specific statistic you want your application to report. Here you have the[ link to the documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/statistics/dds_layer/domainParticipant.html#automatically-enabling-statistics-datawriters) explaining how to do so.  

For this case, my recommendation is to do it via environment variable. On each terminal running a Fast DDS application, just run the following command:

```bash
export FASTDDS_STATISTICS="HISTORY_LATENCY_TOPIC;NETWORK_LATENCY_TOPIC;PUBLICATION_THROUGHPUT_TOPIC;SUBSCRIPTION_THROUGHPUT_TOPIC;RTPS_SENT_TOPIC;RTPS_LOST_TOPIC;HEARTBEAT_COUNT_TOPIC;ACKNACK_COUNT_TOPIC;NACKFRAG_COUNT_TOPIC;GAP_COUNT_TOPIC;DATA_COUNT_TOPIC;RESENT_DATAS_TOPIC;SAMPLE_DATAS_TOPIC;PDP_PACKETS_TOPIC;EDP_PACKETS_TOPIC;DISCOVERY_TOPIC;PHYSICAL_DATA_TOPIC"
```

This will enable all the statistics in Fast DDS applications. In case you want to report just the latency statistic, run this other command:

```bash 
export FASTDDS_STATISTICS="HISTORY_LATENCY_TOPIC"
```

**Comment by wannastar** [comment]
![image](https://github.com/eProsima/Fast-DDS-monitor/assets/42711319/0d267688-9fa4-4304-a33c-5e1634e7ab8e)
![image](https://github.com/eProsima/Fast-DDS-monitor/assets/42711319/3cd6a997-c7cf-4552-aa18-811f5cd7aa99)

yes, i had already stated this variable in ~/.bashrc like this. i can lookup the participant info by fastddsspy link pictur2, so there is  the statisctic info datawriter.

**Reply by wannastar** [reply]
but it is still empty.

**Reply by wannastar** [reply]
DDSHelloworldExample demo can get statistic data, but LatencyTest demo could not.  you can have a try.
